src/Lti/LoadBalancer/LtiInstanceLoadBalancerFactory.php

```diff
@@ -0,0 +1,64 @@
+<?php declare(strict_types=1);
+
+/**
+ *  This program is free software; you can redistribute it and/or
+ *  modify it under the terms of the GNU General Public License
+ *  as published by the Free Software Foundation; under version 2
+ *  of the License (non-upgradable).
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program; if not, write to the Free Software
+ *  Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ *  Copyright (c) 2019 (original work) Open Assessment Technologies S.A.
+ */
+
+namespace App\Lti\LoadBalancer;
+
+use LogicException;
+
+class LtiInstanceLoadBalancerFactory
+{
+    public const LOAD_BALANCER_STRATEGY_USERNAME = 'username';
+    public const LOAD_BALANCER_STRATEGY_USER_GROUP_ID = 'userGroupId';
+
+    private const VALID_LOAD_BALANCER_STRATEGIES = [
+        self::LOAD_BALANCER_STRATEGY_USERNAME,
+        self::LOAD_BALANCER_STRATEGY_USER_GROUP_ID,
+    ];
+
+    /** @var string[] */
+    private $ltiInstances;
+
+    public function __construct(array $ltiInstances)
+    {
+        $this->ltiInstances = $ltiInstances;
+    }
+
+    /**
+     * @throws LogicException
+     */
+    public function __invoke(string $loadBalancerStrategy): LtiInstanceLoadBalancerInterface
+    {
+        switch ($loadBalancerStrategy) {
```

Comment:
But wouldn't it be better to check like `in_array` then using `switch/case` with `default` action?
p.s.
I know that "tastes differ", just curious about it

Reply:
What is the difference?

Reply:
No real difference :) It's more about tastes
p.s.
Plus, just for me `in_array` is a bit easier to read. But, again, it's just a question of taste